src/Grenade/Layers/Convolution.hs

```diff
@@ -105,21 +116,21 @@ instance Show (Convolution c f k k' s s') where
             px = (fmap . fmap . fmap) render ms
         in unlines $ foldl1 (zipWith (\a' b' -> a' ++ "   |   " ++ b')) $ px
 
-randomConvolution :: ( MonadRandom m
-                     , KnownNat channels
-                     , KnownNat filters
-                     , KnownNat kernelRows
-                     , KnownNat kernelColumns
-                     , KnownNat strideRows
-                     , KnownNat strideColumns
-                     , KnownNat kernelFlattened
-                     , kernelFlattened ~ (kernelRows * kernelColumns * channels))
-                  => m (Convolution channels filters kernelRows kernelColumns strideRows strideColumns)
-randomConvolution = do
-    s     <- getRandom
-    let wN = uniformSample s (-1) 1
-        mm = konst 0
+
+instance ( KnownNat c
```

Comment:
this became quite unreadable IMO.  More than 2 type variables? Use long names.

Reply:
fixed